templates/commands/render.go

```diff
@@ -323,6 +347,72 @@ func (r *Render) realRun(ctx context.Context, rp *runParams) (outErr error) {
 	return nil
 }
 
+// processUnknownInputs checks for any unknown input flags and returns them in a slice.
+func (r *Render) processUnknownInputs(spec *model.Spec) []string {
+	unknownInputs := make([]string, 0)
+
+	specInputs := make(map[string]any)
+	for _, v := range spec.Inputs {
+		specInputs[v.Name.Val] = struct{}{}
+	}
+
+	for key := range r.flagInputs {
+		if _, ok := specInputs[key]; !ok {
+			unknownInputs = append(unknownInputs, key)
+		}
+	}
+
+	return unknownInputs
+}
+
+// promptForInputs checks the flag inputs and prompts for missing inputs,
+// using a default value if configured.
+func (r *Render) promptForInputs(ctx context.Context, spec *model.Spec) error {
+	for _, i := range spec.Inputs {
+		_, ok := r.flagInputs[i.Name.Val]
+		if !ok {
+			defaultText := ""
+			if i.Default != nil {
+				defaultText = fmt.Sprintf(" (%s)", i.Default.Val)
+			}
+
+			v, err := r.Prompt(ctx, "Enter value for %s%s:   ", i.Name.Val, defaultText)
```

Comment:
Should we print the description here too (maybe on a preceding line(s))? I could go either way.

Reply:
Created #65 to track this.